Add support for ruby

ric2b opened the pull request:
This was written by Cursor as I have no experience with golang, but I did test it with two gems and was able to generate visualizations that look correct.

I also reviewed the code and compared it to the Python implementation and it looked ok to me.

matzehuels reviewed (changes requested):
Not bad. Can you clean up `rails.json` and `rspec.json` from the `pkg` folder? Just pick one example ruby gem and add the `graph.json` to `examples/real`. Also can you clarify what you mean by
> _I also reviewed the code and compared it to the Python implementation and it looked of to me._

ric2b commented:
Sorry, meant "looked ok", that was a typo.

Updated the PR with rspec as an example and removed the .json that were accidentally included in the previous commit.

ric2b commented on examples/real/rspec.json:
```diff
@@ -0,0 +1,98 @@
+        "license": "MIT",
+        "version": "3.13.7"
+      }
+    },
```
They are a core part of it, the "rspec" testing framework is implemented as several modules that can also be used individually outside of rspec.

You can see them [here](https://rubygems.org/gems/rspec) or in [the documentation](https://rspec.info/documentation/).

matzehuels replied:
Cool, thanks! can you just rebase it (see merge conflicts), then I'll approve and merge it.
